tools/pythonpkg/setup.py

```diff
@@ -380,69 +382,132 @@ def setup_data_files(data_files):
+- tools/pythonpkg/setup.py
+"""
 MAIN_BRANCH_VERSIONING = False
 if os.getenv('MAIN_BRANCH_VERSIONING') == "0":
     MAIN_BRANCH_VERSIONING = False
 if os.getenv('MAIN_BRANCH_VERSIONING') == "1":
     MAIN_BRANCH_VERSIONING = True
 
+""" Regexp to parse versions in PKG-INFO in line with PEP 440. Supports a little more than we need right now
+(pre-, post- and dev-releases), but might come in handy at some point.
+"""
+PKG_INFO_VERSION_RE = re.compile(
+    r"^(?P<version>\d+\.\d+\.\d+)"
+    r"(?P<suffix>(a\d+|b\d+|rc\d+|\.dev\d+|\.post\d+)?)$"
+)
 
-def parse(root: str | Path, config: Configuration) -> ScmVersion | None:
-    from setuptools_scm.git import parse as git_parse
-    from setuptools_scm.version import meta
+""" Regexp to parse versions in OVERRIDE_GIT_DESCRIBE (i.e. git describe --tags output).Supports a little more than we
+need right now (pre-, post- and dev-releases), but might come in handy at some point.
+
+Assumptions:
+- The format is <version_tag>(-(dev)?<distance>(-<node>)?)?
+- <version_tag> is required and must start with 'v', followed by a PEP 440-compliant version:
+  e.g. v1.2.3, v2.0.0a1, v1.2.3.post1, v1.2.3.dev2
+- <distance> is optional and may optionally be prefixed with "dev" (which is not in line with git describe output
+  but seems to be used here and there...).
+- <node> is optional and represents the Git commit hash (usually prefixed by "g").
+"""
+GIT_DESCRIBE_RE = re.compile(
+    r"""^
+    (?P<version_tag>                       # Full version tag (starting with v)
+        v[0-9]+\.[0-9]+\.[0-9]+            # Core: v1.2.3
+        (?:                               
+            (?:a|b|rc)[0-9]+ |             # Pre-release: a1, b2, rc3
+            \.post[0-9]+ |                 # Post-release: .post1
+            \.dev[0-9]+                    # Dev-release: .dev2
+        )?
+    )
+    (?:
+        -(?:dev)?(?P<distance>[0-9]+)          # Optional: -4 or -dev4
+        (?:-g?(?P<node>[0-9a-fA-F]+))?         # Optional: -g<hash>
+    )?
+    $
+    """,
+    re.VERBOSE
+)
 
-    override = os.getenv('OVERRIDE_GIT_DESCRIBE')
-    if override:
-        parts = override.split('-')
-        if len(parts) == 3:
-            # Already in correct format tag-distance-gnode
-            tag, distance, node = parts
-            return meta(tag=tag, distance=int(distance), node=node[1:], config=config)
-        elif len(parts) == 1:
-            # Just tag, add -0-gnode
-            tag = parts[0]
-            distance = 0
-        elif len(parts) == 2:
-            # tag-distance, need to add -gnode
-            tag, distance = parts
-        else:
-            raise ValueError(f"Invalid OVERRIDE_GIT_DESCRIBE format: {override}")
-        return meta(tag=tag, distance=int(distance), config=config)
+def git_describe():
+    """ Get the git describe string either from the remote repo or from OVERRIDE_GIT_DESCRIBE.
 
-    versioning_tag_match = 'v*.*.0' if MAIN_BRANCH_VERSIONING else 'v*.*.*'
```

Comment:
I think some version of this needs to be kept around

Reply:
Yes that's right, it still lives in the `version_scheme` but it'll have to move into the `parse()` function, I think. Just want to get the parsing right first.